cmd/clairctl/clair/clair.go

```diff
@@ -0,0 +1,289 @@
+func (feature Feature) Weight() int {
+	weight := 0
+	
+	for _, v := range feature.Vulnerabilities {
+		weight += v.Weight()
+	}
+	
+	return weight
+}
+
+// VulnerabilitiesBySeverity sorting vulnerabilities by severity
+type VulnerabilitiesBySeverity []Vulnerability
+
+func (a VulnerabilitiesBySeverity) Len() int { return len(a) }
+func (a VulnerabilitiesBySeverity) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a VulnerabilitiesBySeverity) Less(i, j int) bool  {
+	return a[i].Weight() > a[j].Weight()
+}
+
+// LayerByVulnerabilities sorting of layers by global vulnerability
+type LayerByVulnerabilities []Layer
+
+func (a LayerByVulnerabilities) Len() int { return len(a) }
+func (a LayerByVulnerabilities) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a LayerByVulnerabilities) Less(i, j int) bool {
+	firstVulnerabilities := 0
+	secondVulnerabilities := 0
+	
+	for _, l := range a[i].Features {
+		firstVulnerabilities = firstVulnerabilities + l.Weight()
+	}
+	
+	for _ , l := range a[j].Features {
+		secondVulnerabilities = secondVulnerabilities + l.Weight()
+	}
+	
+	return firstVulnerabilities > secondVulnerabilities
+}
+
+// FeatureByVulnerabilities sorting off features by vulnerabilities
+type FeatureByVulnerabilities []Feature
+
+func (a FeatureByVulnerabilities) Len() int { return len(a) }
+func (a FeatureByVulnerabilities) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+
+func (a FeatureByVulnerabilities) Less(i, j int) bool {
+	return a[i].Weight() > a[j].Weight()
+}
+
+// SortLayers give layers ordered by vulnerability algorithm
+func (imageAnalysis ImageAnalysis) SortLayers() []Layer {
+	layers := []Layer{}
+	
+	for _, l := range imageAnalysis.Layers {
+		features := []Feature{}
+		
+		for _, f := range l.Layer.Features {
+			vulnerabilities := []Vulnerability{}
+			
+			for _, v := range f.Vulnerabilities {
+				nv := Vulnerability{
+					Name:        v.Name,
+					Severity:    v.Severity,
+					IntroduceBy: f.Name + ":" + f.Version,
+					Description: v.Description,
+					Layer:       l.Layer.Name,
+					Link:        v.Link,
+				}
+				
+				vulnerabilities = append(vulnerabilities, nv);
+			}
+			
+			sort.Sort(VulnerabilitiesBySeverity(vulnerabilities))
+			
+			nf := Feature{
+				Name: f.Name,
+				Version: f.Version,
+				Vulnerabilities: vulnerabilities,
+			}
+			
+			features = append(features, nf);
+		}
+		
+		sort.Sort(FeatureByVulnerabilities(features))
+		
+		nl := Layer{
+			Name: l.Layer.Name,
+			Path: l.Layer.Path,
+			Features: features,
+		}
+		layers = append(layers, nl);
+	}
+	
+	sort.Sort(LayerByVulnerabilities(layers));
+	
+	return layers;
+}
+
+// SortVulnerabilities get all vulnerabilities sorted by Severity
+func (imageAnalysis ImageAnalysis) SortVulnerabilities() []Vulnerability {
+	vulnerabilities := []Vulnerability{}
+	
+	// there should be a better method, but I don't know how to easlily concert []v1.Vulnerability to [Vulnerability]
+	for _, l := range imageAnalysis.Layers {
```

Comment:
Again, this shouldn't be aggregated. The Clair's API gives all the vulnerabilities for an image properly if you query with the latest layer.